app/controllers/application.js

```diff
@@ -23,6 +24,20 @@ export default class ApplicationController extends Controller {
 
   searchTerms = '';
 
+  popupFeature = false;
+
+  popupLocation = null;
+
+  popupWfParkId = null;
+
+  popupParkName = null;
+
+  popupAgency = null;
+
+  popupStatus = null;
+
+  popupLink = null;
```

Comment:
I think the namespacing here ("popup{Something}") signals the need for a new component that handles this state/functionality. There's a tendency to put everything in the application controller, which makes it long and unwieldy.